mne/io/base.py

```diff
@@ -363,6 +369,35 @@ def __init__(self, info, preload=False,
         self._comp = None
         self._filenames = list(filenames)
         self.orig_format = orig_format
+        # Sanity check and set original units, if provided by the reader:
+        if orig_units:
+            if not isinstance(orig_units, dict):
+                raise ValueError('orig_units must be of type dict, but got '
+                                 ' {}'.format(type(orig_units)))
+
+            # original units need to be truncated to 15 chars, which is what
+            # the MNE IO procedure also does with the other channels
+            orig_units_trunc = [ch[:15] for ch in orig_units]
+
+            # STI 014 channel is native only to fif ... for all other formats
+            # this was artificially added by the IO procedure, so remove it
+            ch_names = list(info['ch_names'])
+            if ('STI 014' in ch_names) and not \
+               (self.filenames[0].endswith('.fif')):
+                ch_names.remove('STI 014')
```

Comment:
I don't understand this. Why are we removing the stim channel here? wouldn't this have collateral effects? I would let it be until we remove it for good.

Reply:
it is only being removed for the sake of the check, it stays in the data after the check.

for the check however, it needs to be removed because we do not have an "original unit" for the stim channel if it was artificially added during the IO procedure (which is always the case except for fif)

Reply:
shouldn't it then treat in the check? rather than modifying the raw object?

Reply:
I don't know what you mean, could you provide an example? Where am I modifying a raw object here?

Reply:
I am working on a copy of the ch_names, see line 384

Reply:
yes, you are completely right. you are working with a copy. my bad.